packages/content/src/types.ts

```diff
@@ -0,0 +1,137 @@
+import { IItem, IGroup } from "@esri/arcgis-rest-types";
+
+export type HubType = 'document' | 'dataset' | 'map' | 'app' | 'site' | 'initiative' | 'event' | 'template';
+
+// Based on the [Hub Content UI Inventory](https://app.lucidchart.com/invitations/accept/dab82a25-2ed9-4f63-8075-3db8873087e3)
+// Attempting a simple _Hungarian Notation_ for attributes suffix: `*Date`, `*Id`, `*Number`. Plural (e.g. `tags`) is an array.
+
+export interface IContent extends IItem {
+  
+  // Inherited metadata (noting relevant subset of all item attributes)
+  // id: string; // item.id
+  // title: string; // item.title
+  // snippet: string; // item.snippet
+  // description: string; // item.description
+  // categories?: Array<string>; // item.categories
+  // tags: Array<string>; // item.tags  
+  // culture: string; // written locale e.g. en, es, cn
+  // url?: string;
+  // size: number;
+
+  // Derived metadata
+  hubType: HubType;
+  maintainer: IHubMember; // item.owner with more user metadata
+  access: { // overrides item.access with more attributes. could flatten.
+    visibility: 'private' | 'org' | 'public'; // item.access
+    permission?: 'view' | 'edit' | 'admin'; // item.itemControl
+    groups?: Array<IGroup>; // item.sharing.groups via content/users/:username/items/:id
+  };
+
+  // Explicit data information since this is a common confusion + bug report
+  createdDate: Date; // formal metadata || new Date(item.created)
+  createdDateSource: string; // description of what was used for this attribute
+  publishedDate: Date; // formal metadata || new Date(item.created)
+  publishedDateSource: string; // description of what was used for this attribute
+  updatedDate: Date; // formal metadata || new Date(item.modified)
+  updatedDateSource: string; // description of what was used for this attribute
+  
+  contentUrl: string; // Link to the raw content. item.url in most (but not all) item types
+  thumbnailUrl: string; // Full URL. item.thumbnail with host + path 
+  
+  license: IHubLicense; // [Future] item.licenseInfo 
+  boundary: IHubGeography; // [Future] Inline, or boolean when stored at known location /resources/boundary.json
+  
+  // Additional metadata from custom/formal elements
+  metadata?: {
+      // Unique or additional formal metadata that will be displayed in sidebar
+  };
```

Comment:
should this be defined as `metadata?: {[field: string]: any}`?

Reply:
The proposal is that it's nested and based on [mdJSON example](https://github.com/adiwg/mdJson-schemas/blob/master/examples/resourceInfo.json)

Reply:
So is it actually an array?

Reply:
No - that link is an array because it's a set of examples. 

A full metadata is an object like [this example](https://gist.github.com/ajturner/d68270883a1fc729e6319d7fc5e5a4f2)

Looking at that makes a point that the full metadata.json would be a sidecar resource. The purpose of this `content.metadata` was to have a configured summary of "Maintainer wants this metadata to appear in result cards / about views". For example, some orgs might want their department phone number and hours to show up. Another might want the methodology or sensor type. 

We can pull this attribute for now from the baseline, with expectation that it would be added in the relatively near future with specifics on what it stores and can we be explicit about its schema.

Reply:
I think Mike is saying that from a typescript perspective, this should be `metadata?: {[field: string]: any}` which means "accept an arbitrary hash"

Reply:
> We can pull this attribute for now from the baseline, with expectation that it would be added in the relatively near future with specifics on what it stores and can we be explicit about its schema.

sounds good to me